internal/gengapic/genrest.go

```diff
@@ -295,14 +360,15 @@ func (g *generator) generateURLString(m *descriptor.MethodDescriptorProto) error
 		requestObject = "body"
 	}
 
-	fmtStr := fmt.Sprintf("%%s%s", info.url)
+	fmtStr := fmt.Sprintf("%s", info.url)
 	// TODO(dovs): handle more complex path urls involving = and *,
 	// e.g. v1beta1/repeat/{info.f_string=first/*}/{info.f_child.f_string=second/**}:pathtrailingresource
 	re := regexp.MustCompile(`{([a-zA-Z0-9_.]+?)}`)
 	fmtStr = re.ReplaceAllStringFunc(fmtStr, func(s string) string { return "%v" })
 
-	p(`url := fmt.Sprintf("%s",`, fmtStr)
-	p("c.endpoint,")
+	// TODO(dovs): handle error
+	p("baseUrl, _ := url.Parse(c.endpoint)")
+	p(`baseUrl.Path += fmt.Sprintf("%s",`, fmtStr)
```

Comment:
Are we guaranteed to have such params? if the regex matching comes up with nothing, this becomes uncompilable code.

Reply:
I don't believe that's the case. If the regex comes up with nothing, then the `ReplaceAllStringFunc` invocation won't add any format string markers, and the result of `FindAllStringSubmatch` will be an empty slice.

Reply:
I meant that the generated code becomes uncompilable, it would look like this: https://play.golang.org/p/1Q8RTVsgSg_U

Reply:
Took me a little while to understand, but found a workaround.